code/pages/SubsitesVirtualPage.php

```diff
@@ -1,45 +1,71 @@
 <?php
+
+namespace SilverStripe\Subsites\Pages;
+
+
+use SilverStripe\CMS\Model\SiteTree;
+use SilverStripe\CMS\Model\VirtualPage;
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\Session;
+use SilverStripe\Core\Config\Config;
+use SilverStripe\Forms\DropdownField;
+use SilverStripe\Forms\LabelField;
+use SilverStripe\Forms\TextareaField;
+use SilverStripe\Forms\TextField;
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\Subsites\Forms\SubsitesTreeDropdownField;
+use SilverStripe\Subsites\Model\Subsite;
+use SilverStripe\View\ArrayData;
+
 class SubsitesVirtualPage extends VirtualPage
 {
+
+    private static $table_name = 'SubsitesVirtualPage';
+
     private static $description = 'Displays the content of a page on another subsite';
 
-    private static $db = array(
+    private static $db = [
         'CustomMetaTitle' => 'Varchar(255)',
         'CustomMetaKeywords' => 'Varchar(255)',
         'CustomMetaDescription' => 'Text',
         'CustomExtraMeta' => 'HTMLText'
-    );
+    ];
+
+    private static $non_virtual_fields = [
+        'SubsiteID'
+    ];
 
     public function getCMSFields()
     {
         $fields = parent::getCMSFields();
 
-        $subsites = DataObject::get('Subsite');
+        $subsites = DataObject::get(Subsite::class);
         if (!$subsites) {
             $subsites = new ArrayList();
         } else {
-            $subsites=ArrayList::create($subsites->toArray());
+            $subsites = ArrayList::create($subsites->toArray());
         }
 
-        $subsites->push(new ArrayData(array('Title' => 'Main site', 'ID' => 0)));
+        $subsites->push(new ArrayData(['Title' => 'Main site', 'ID' => 0]));
 
         $fields->addFieldToTab(
             'Root.Main',
             DropdownField::create(
-                "CopyContentFromID_SubsiteID",
-                _t('SubsitesVirtualPage.SubsiteField', "Subsite"),
+                'CopyContentFromID_SubsiteID',
+                _t('SubsitesVirtualPage.SubsiteField', 'Subsite'),
                 $subsites->map('ID', 'Title')
             )->addExtraClass('subsitestreedropdownfield-chooser no-change-track'),
             'CopyContentFromID'
         );
 
         // Setup the linking to the original page.
         $pageSelectionField = new SubsitesTreeDropdownField(
-            "CopyContentFromID",
-            _t('VirtualPage.CHOOSE', "Choose a page to link to"),
-            "SiteTree",
-            "ID",
-            "MenuTitle"
+            'CopyContentFromID',
+            _t('VirtualPage.CHOOSE', 'Choose a page to link to'),
```

Comment:
This has been namespaced now, so will be `_t('SilverStripe\\CMS\\Model\\VirtualPage.CHOOSE', 'Linked Page')`. Note that `VirtualPage::class` won't work with the framework's current text collector, so we have to be more explicit with FCQNs that aren't just `__CLASS__`